external-crates/move/crates/move-vm-runtime/src/execution/tracing/tracer.rs

```diff
@@ -320,7 +320,9 @@ impl VMTracer<'_> {
         machine: &MachineState,
         stack_idx: usize,
     ) -> Option<TraceValue> {
+        // TODO: Add an assertion establishing that operand_stack and type_stack are in sync.
```

Comment:
Not sure if we need these assertions, as we assert this in the only cases where we perform stack transformations and the stacks should line up -- at the beginning of `start_instruction_impl` and at the end of `end_instruction_impl`). In these cases, I'd probably prefer to just output a trace error like we do right now in these cases.